Yesod/Auth/OAuth2/Salesforce.hs

```diff
@@ -84,19 +85,21 @@ oauth2SalesforceSandboxScoped scopes clientId clientSecret =
     authOAuth2 svcNameSb oauth fetchSalesforceSandboxUser
   where
     oauth = OAuth2
-        { oauthClientId            = encodeUtf8 clientId
-        , oauthClientSecret        = encodeUtf8 clientSecret
-        , oauthOAuthorizeEndpoint  = encodeUtf8 $ "https://test.salesforce.com/services/oauth2/authorize?scope=" <> T.intercalate " " scopes
+        { oauthClientId            = clientId
+        , oauthClientSecret        = clientSecret
+        , oauthOAuthorizeEndpoint  = "https://test.salesforce.com/services/oauth2/authorize" `withQuery`
+            [ ("scope=", encodeUtf8 $ T.intercalate " " scopes)
```

Comment:
should `"scope="` be `"scope"`?

Reply:
Taking this one step further - thoughts on extracting tuple generation for scope to be shared across each of these?

Reply:
Nice catch. I think pulling out a function, while useful for the cases *only* adding `scope`, will make the code where you're adding more than `scope` more cumbersome.

Reply:
You were right, I'm extracting `scopeParam` :)